src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl

```diff
@@ -147,8 +147,8 @@ end
 sets the initial value for state variables
 dispatches to ocean_init_state! which is defined in a problem file such as SimpleBoxProblem.jl
 """
-function init_state_prognostic!(m::HBModel, Q::Vars, A::Vars, coords, t)
-    return ocean_init_state!(m, m.problem, Q, A, coords, t)
+function init_state_prognostic!(m::HBModel, Q::Vars, A::Vars, localgeo, t)
+    return ocean_init_state!(m, m.problem, Q, A, localgeo.coord, t)
```

Comment:
Could you actually pass `localgeo` into `ocean_init_state!` as well? And then just extract the `coords` the same way as in the `ocean_init_aux!` function? And same thing for the ShallowWaterModel and OceanModel?

It's more work on your end, but gets the interfaces looking really nice.

Reply:
Done. Let me know what you think.